miner/worker_test.go

```diff
@@ -680,15 +744,18 @@ func TestL1MsgCorrectOrder(t *testing.T) {
 	chainConfig.Clique = &params.CliqueConfig{Period: 1, Epoch: 30000}
 	engine = clique.New(chainConfig.Clique, db)
 
+	maxTxPerBlock := 4
+	chainConfig.Scroll.MaxTxPerBlock = &maxTxPerBlock
 	chainConfig.Scroll.L1Config = &params.L1Config{
 		NumL1MessagesPerBlock: 10,
 	}
 
-	// Insert 3 l1msgs
+	// Insert 4 l1msgs, with one be skipped.
```

Comment:
Why change this test? Ideally one test should test one behaviour

Reply:
because `4 >= l2 msg + l1 non-skip msg && 5 < l2 msg + l1 non-skip msg + l1 skip msg`
so the 4 txs are acceptable.

Reply:
reverted in [9f74baf](https://github.com/scroll-tech/go-ethereum/pull/486/commits/9f74baf2fe0afe6a218055f948628f6c273f5a5a)